content/en/blog/2023/egress-sni/index.md

```diff
@@ -0,0 +1,348 @@
+
+In order to deploy the sample configuration, create the `istio-egress` namespace, allow injection in it and deploy the following manifest.
+
+{{< text yaml >}}
+# New k8s cluster service to put egressgateway into the Service Registry
+# so application sidecars can route traffic towards it within the mesh.
+apiVersion: v1
+kind: Service
+metadata:
+  name: egressgateway
+  namespace: istio-egress
+spec:
+  type: ClusterIP
+  selector:
+    istio: egressgateway
+  ports:
+  - port: 443
+    name: tls-egress
+    targetPort: 8443
+---
+# Gateway deployment with injection method
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-egressgateway
+  namespace: istio-egress
+spec:
+  selector:
+    matchLabels:
+      istio: egressgateway
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels:
+        istio: egressgateway
+        sidecar.istio.io/inject: "true"
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto # The image will automatically update each time the pod starts.
+        securityContext:
+          capabilities:
+            drop:
+            - ALL
+          runAsUser: 1337
+          runAsGroup: 1337
+---
+# Set up roles to allow reading credentials for TLS
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: istio-egressgateway-sds
+  namespace: istio-egress
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "watch", "list"]
+- apiGroups:
+  - security.openshift.io
+  resourceNames:
+  - anyuid
+  resources:
+  - securitycontextconstraints
+  verbs:
+  - use
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: istio-egressgateway-sds
+  namespace: istio-egress
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: istio-egressgateway-sds
+subjects:
+- kind: ServiceAccount
+  name: default
+---
+apiVersion: v1
+kind: List
+items:
+# Define a new listener that enforces Istio mTLS on inbound connections.
+# This is where sidecar will route the application traffic, wrapped into
+# Istio mTLS.
+  - apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: egressgateway
+      namespace: istio-system
+    spec:
+      selector:
+        istio: egressgateway
+      servers:
+      - port:
+          number: 8443
+          name: tls-egress
+          protocol: TLS
+        hosts: 
+          - "*"
+        tls:
+          mode: ISTIO_MUTUAL
+# VirtualService that will instruct sidecars in the mesh to route the outgoing
+# traffic to the egress gateway Service if the SNI target hostname matches
+  - apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: direct-wildcard-through-egress-gateway
+      namespace: istio-system
+    spec:
+      hosts:
+        - "*.foo.com"
```

Comment:
The problem with wikipedia is that I could not find any second service on their domains which are served by a different set of IPs, so the use-case is not valid with it unfortunately 🤷 . Not sure what else should I use which is generic enough for an example. I could certainly use IBM targets for example...

Reply:
True that it's not needed for wekipedia, but wikipedia can still be used to show the config is set up properly and working, and it shows that it can be used for any wildcard host without needing to care about what underlying servers are involved. The old doc used wikipedia to demonstrate. That said, you could use something as general as `*.com` and show that you can access `google.com` and `ibm.com` to demonstrate, but if you want a more realistic example, I don't see anything wrong with just picking a couple `ibm.com` subdomains to demonstrate.